### PR TITLE
feat(live_filter): filter nodes from loaded buffers

### PR DESCRIPTION
**Pull request by kyazdani42**
Breaking: changing `f` binding to `ff` for the input type filter
Adding `fb` for buffer type filter.

Note: if folders are not shown (`:help nvim-tree.live_filter.always_show_folders`), the filter will not find the nodes since these will be filtered out. 
This option will only work well with find_files enabled and always_show_folders true.

We need to improve the filter algorithm to be able to search recursively. The UI can also be improved for the buffer filtering, something like displaying the buffer number. But this will come in other PRs

### CODE REVIEW

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -1062,7 +1066,8 @@ DEFAULT MAPPINGS                                     *nvim-tree-default-mappings
     { key = "]c",                             action = "next_git_item" }
     { key = "-",                              action = "dir_up" }
     { key = "s",                              action = "system_open" }
-    { key = "f",                              action = "live_filter" }
+    { key = "ff",                             action = "live_filter" }
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:1069]: On pressing ff when fb is active, the filter changes text to `/Buffer/` which is not useful.

**Comment by alex-courtis** [comment]
> We need to improve the filter algorithm to be able to search recursively.

I'd like that for regular filter. Out of scope here.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Functionality is useful, works.

The UX is not great. We do not need to present this functionality to the user as a filter, even if we do it that way internally.

Proposal:
- leave existing `f` and `F` unchanged
  - avoids breaking change
- use `B` to toggle buffer filtering, like `H` `I` etc
  - `B` will not show `[FILTER]: /Buffer/` as this is confusing
- `f` and `B` may be run together, like `f` and `H`

There is no indication that `B` is on, but that is OK as there is no indication that `H` is on. Users appear happy with that.

**Comment by kyazdani42** [comment]
After your comments, i wonder if this feature should even be inside the live-filter :smile: I think we could just implement this filter on the renderer since we already check for opened buffers there.

**Comment by kyazdani42** [comment]
Also this might be a good idea to do like neo-tree and display the buf numbers next to the filenames.
I'm closing in favor of a better implementation.

**Pull request closed without merging**